core/iwasm/common/wasm_runtime_common.c

```diff
@@ -315,7 +319,18 @@ wasm_runtime_get_exec_env_tls()
 {
     return exec_env_tls;
 }
-#endif /* end of OS_ENABLE_HW_BOUND_CHECK */
+#endif
+
+#ifdef OS_ENABLE_BLOCK_INSN_INTERRUPT
+static void
+interrupt_block_insn_sig_handler()
+{
+    WASMJmpBuf *jmpbuf_node = exec_env_tls->jmpbuf_stack_top;
```

Comment:
jmpbuf_stack_top here can be for another handler.

Reply:
Yes, if HW bound check is enabled, it will get a "HW bound" `jmpbuf` https://github.com/bytecodealliance/wasm-micro-runtime/pull/1930#discussion_r1096389777, but that should not be a problem unless I'm missing something

Reply:
you can even get a signal between wasm_exec_env_push_jmpbuf and os_setjmp.

Reply:
> you can even get a signal between wasm_exec_env_push_jmpbuf and os_setjmp.

In that case, the `jmpbuf` would be invalid, as it is in general before being set up by `os_setjmp`. Isn't that the same for the HW bound check before my PR?

I wonder if we have to disable/re-enable the signals to avoid the signal handler being called in the middle. We want to avoid signal handlers being called between `wasm_exec_env_push_jmpbuf` and `os_setjmp` as they would use a non-initialized `jmpbuf` iiuc.

Reply:
it isn't a problem for hw bound check because the signal for it is basically synchronous to the thread execution. ie. the thread itself can control where it can happen.

Reply:
Can we maybe invert the order of `wasm_exec_env_push_jmpbuf` and `os_setjmp`? So that the `jmpbuf` is only pushed after being initialized by `os_setjmp`

Reply:
> you can even get a signal between wasm_exec_env_push_jmpbuf and os_setjmp.

Apart from this, do you think there are other cases that are not covered?